media-libs/osl/osl-1.9.5.recipe

```diff
@@ -0,0 +1,154 @@
+	cmd:testshade$commandSuffix
+	cmd:testshade_dso$commandSuffix
+	lib:liboslcomp$secondaryArchSuffix = $portVersion compat >= 1
+	lib:liboslexec$secondaryArchSuffix = $portVersion compat >= 1
+	lib:liboslnoise$secondaryArchSuffix = $portVersion compat >= 1
+	lib:liboslquery$secondaryArchSuffix = $portVersion compat >= 1
+	lib:libtestshade$secondaryArchSuffix = $portVersion compat >= 1
+	lib:osl.imageio$secondaryArchSuffix = $portVersion compat >= 1
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libboost_atomic$secondaryArchSuffix
+	lib:libboost_chrono$secondaryArchSuffix
+	lib:libboost_date_time$secondaryArchSuffix
+	lib:libboost_filesystem$secondaryArchSuffix
+	lib:libboost_serialization$secondaryArchSuffix
+	lib:libboost_system$secondaryArchSuffix
+	lib:libboost_thread$secondaryArchSuffix
+	lib:libboost_wave$secondaryArchSuffix
+	lib:libHalf$secondaryArchSuffix
+	lib:libIex_2_2$secondaryArchSuffix
+	lib:libilmimf$secondaryArchSuffix
+	lib:libIlmThread_2_2$secondaryArchSuffix
+	lib:libImath_2_2$secondaryArchSuffix
+	lib:libLLVM_5.0$secondaryArchSuffix
+	lib:libopenimageio$secondaryArchSuffix
+	lib:libpugixml$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	osl${secondaryArchSuffix}_devel = $portVersion compat >= 1
+	devel:liboslcomp$secondaryArchSuffix = $portVersion compat >= 1
+	devel:liboslexec$secondaryArchSuffix = $portVersion compat >= 1
+	devel:liboslnoise$secondaryArchSuffix = $portVersion compat >= 1
+	devel:liboslquery$secondaryArchSuffix = $portVersion compat >= 1
+	devel:libtestshade$secondaryArchSuffix = $portVersion compat >= 1
+	devel:osl.imageio$secondaryArchSuffix = $portVersion compat >= 1
+	"
+REQUIRES_devel="
+	osl$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	gcc_syslibs_devel$secondaryArchSuffix
+	devel:libboost_atomic$secondaryArchSuffix
+	devel:libboost_chrono$secondaryArchSuffix
+	devel:libboost_date_time$secondaryArchSuffix
+	devel:libboost_filesystem$secondaryArchSuffix
+	devel:libboost_serialization$secondaryArchSuffix
+	devel:libboost_system$secondaryArchSuffix
+	devel:libboost_thread$secondaryArchSuffix
+	devel:libboost_wave$secondaryArchSuffix
+	devel:libHalf$secondaryArchSuffix
+	devel:libIex_2_2$secondaryArchSuffix
+	devel:libilmimf$secondaryArchSuffix
+	devel:libIlmThread_2_2$secondaryArchSuffix
+	devel:libImath_2_2$secondaryArchSuffix
+	devel:libLLVM_5.0$secondaryArchSuffix
+	devel:libpartio$secondaryArchSuffix
+	devel:libopenimageio$secondaryArchSuffix
+	devel:libpugixml$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:bison
+	cmd:clang$secondaryArchSuffix
+	cmd:cmake
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python2
+	"
+
+defineDebugInfoPackage osl$secondaryArchSuffix \
+	$libDir/liboslcomp.so \
+	$libDir/liboslexec.so \
+	$libDir/liboslnoise.so \
+	$libDir/liboslquery.so \
+	$libDir/libtestshade.so \
+	$libDir/osl.imageio.so \
```

Comment:
I haven't build this package, but if they don't have any `SOVERSION`, you could drop the entire `compat` part, as there aren't any kind of ABI stability for this kind of library (or the ABI is constant and never changes, but not really likely)